src/util/mathematical_expr.cpp

```diff
@@ -7,3 +7,17 @@ Author: Daniel Kroening, kroening@kroening.com
 \*******************************************************************/
 
 #include "mathematical_expr.h"
+#include "mathematical_types.h"
+
+function_application_exprt::function_application_exprt(
+  const symbol_exprt &_function,
+  argumentst _arguments)
+  : binary_exprt(
+      _function,
+      ID_function_application,
+      multi_ary_exprt(irep_idt(), std::move(_arguments), typet()),
```

Comment:
This looks very odd. Can we use `ID_arguments`? `exprt{ID_arguments, std::move(_arguments)}` would IMHO look cleaner. (But that's probably not possible until #3502 is merged.)

Reply:
Sure, could use ID_arguments, but could also contemplate something like ID_tuple (which is what is happening here).

Reply:
(Either should be a separate PR, and would need to change all constructors).